pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -51,52 +52,73 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
-				rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceMemory] = *rsrc
+			if len(rs.PodSpec.Containers) > 1 {
+				rs.PodSpec.Containers[idx].Name = cfg.Defaults.UserContainerName(ctx) + "-" + strconv.Itoa(idx)
+			} else {
+				rs.PodSpec.Containers[idx].Name = cfg.Defaults.UserContainerName(ctx)
 			}
 		}
 
-		if rs.PodSpec.Containers[idx].Resources.Limits == nil {
-			rs.PodSpec.Containers[idx].Resources.Limits = corev1.ResourceList{}
-		}
-		if _, ok := rs.PodSpec.Containers[idx].Resources.Limits[corev1.ResourceCPU]; !ok {
-			if rsrc := cfg.Defaults.RevisionCPULimit; rsrc != nil {
-				rs.PodSpec.Containers[idx].Resources.Limits[corev1.ResourceCPU] = *rsrc
-			}
+		rs.applyDefault(&rs.PodSpec.Containers[idx], cfg)
+	}
+}
+func (rs *RevisionSpec) applyDefault(container *corev1.Container, cfg *config.Config) {
+	if container.Resources.Requests == nil {
+		container.Resources.Requests = corev1.ResourceList{}
+	}
+	if _, ok := container.Resources.Requests[corev1.ResourceCPU]; !ok {
+		if rc := cfg.Defaults.RevisionCPURequest; rc != nil {
+			container.Resources.Requests[corev1.ResourceCPU] = *rc
 		}
-		if _, ok := rs.PodSpec.Containers[idx].Resources.Limits[corev1.ResourceMemory]; !ok {
-			if rsrc := cfg.Defaults.RevisionMemoryLimit; rsrc != nil {
-				rs.PodSpec.Containers[idx].Resources.Limits[corev1.ResourceMemory] = *rsrc
-			}
+	}
+	if _, ok := container.Resources.Requests[corev1.ResourceMemory]; !ok {
+		if rm := cfg.Defaults.RevisionMemoryRequest; rm != nil {
+			container.Resources.Requests[corev1.ResourceMemory] = *rm
 		}
-		if rs.PodSpec.Containers[idx].ReadinessProbe == nil {
-			rs.PodSpec.Containers[idx].ReadinessProbe = &corev1.Probe{}
+	}
+
+	if container.Resources.Limits == nil {
+		container.Resources.Limits = corev1.ResourceList{}
+	}
+	if _, ok := container.Resources.Limits[corev1.ResourceCPU]; !ok {
+		if rc := cfg.Defaults.RevisionCPULimit; rc != nil {
+			container.Resources.Limits[corev1.ResourceCPU] = *rc
 		}
-		if rs.PodSpec.Containers[idx].ReadinessProbe.TCPSocket == nil &&
-			rs.PodSpec.Containers[idx].ReadinessProbe.HTTPGet == nil &&
-			rs.PodSpec.Containers[idx].ReadinessProbe.Exec == nil {
-			rs.PodSpec.Containers[idx].ReadinessProbe.TCPSocket = &corev1.TCPSocketAction{}
+	}
+	if _, ok := container.Resources.Limits[corev1.ResourceMemory]; !ok {
+		if rm := cfg.Defaults.RevisionMemoryLimit; rm != nil {
+			container.Resources.Limits[corev1.ResourceMemory] = *rm
 		}
+	}
 
-		if rs.PodSpec.Containers[idx].ReadinessProbe.SuccessThreshold == 0 {
-			rs.PodSpec.Containers[idx].ReadinessProbe.SuccessThreshold = 1
+	if len(rs.PodSpec.Containers) == 1 {
+		rs.applyProbes(container)
+	} else {
+		// If there are multiple containers then default probes will be applied to the container where user specified PORT
+		// default probes will not be applied for non serving containers
+		if len(container.Ports) != 0 {
+			rs.applyProbes(container)
 		}
+	}
```

Comment:
```suggestion
		// If there are multiple containers then default probes will be applied to the container where user specified PORT
		// default probes will not be applied for non serving containers
		if len(container.Ports) != 0 {
			rs.applyProbes(container)
		}
```
we already validated that only a single container will have port.

Reply:
here i am not doing port validation.

Instead when multiple containers specified probes default values applied to only that particular container where port specified.
For remaining(Non Serving) containers default probes should not apply

Reply:
I don't see how this contradicts what I said :-)

Reply:
I am confused :smile: 
I will change it